src/main/java/sirius/biz/tenants/deletion/TenantDeletionWorkItem.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.tenants.deletion;
+
+import sirius.biz.process.logs.ProcessLog;
+import sirius.biz.tenants.Tenant;
+import sirius.kernel.di.std.Priorized;
+
+/**
+ * Objects of this interface will be called when a tenant is deleted and are responsible for deleting external tenant
+ * data, e.g {@link sirius.biz.tenants.UserAccount UserAccounts} in {@link DeleteUserAccountsWorkItem}.
+ */
+public interface TenantDeletionWorkItem extends Priorized {
+
+    /**
+     * Called with the {@link Tenant} for which data should be deleted.
+     *
+     * @param tenant   the tenant which is deleted
+     * @param simulate wheter the deleting should only be simulated
+     * @return A {@link ProcessLog} message, whichis logged by the executing job
+     */
+    ProcessLog execute(Tenant<?> tenant, boolean simulate);
+
+    /**
+     * Returns the message key for the {@link ProcessLog} returned by the {@link #execute(Tenant, boolean)} method.
+     *
+     * @param simulate wheter the deleting should only be simulated
+     * @return the message key for the log message
+     */
+    default String getMessageKey(boolean simulate) {
```

Comment:
no default methods in new code - this is a Java Hack to enhance old interfaces..